test/promise.js

```diff
@@ -0,0 +1,88 @@
+
+var Router = require('..')
+var utils = require('./support/utils')
+var Promise = require('bluebird')
+
+var assert = utils.assert
+var createServer = utils.createServer
+var request = utils.request
+
+describe('Promise', function () {
+  it('rejecting will trigger error handlers', function (done) {
+    var router = Router()
+    var server = createServer(router)
+
+    router.use(function (req, res) {
+      return new Promise(function (resolve, reject) {
+        reject(new Error('Happy error'))
+      })
+    })
+
+    request(server)
+    .get('/')
+    .expect(500, done)
+  })
+
+  it('will be ignored if next is called', function (done) {
+    var router = Router()
+    var server = createServer(router)
+    var count = 0
+
+    router.use(function (req, res, next) {
+      count++
```

Comment:
In that case, simply call `done(new Error('unexpected call'))` in the thing that should be called instead of trying to do the counting (because it could still end up getting called, but after you checked the count and then your test would incorrectly pass).